src/components/Ruler/index.js

```diff
@@ -0,0 +1,93 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import RulerTooltip from '../RulerTooltip';
+import { pointPropType, rulerPropType } from '../../utils/proptypes';
+
+const propTypes = {
+  width: PropTypes.number.isRequired,
+  height: PropTypes.number.isRequired,
+  points: PropTypes.arrayOf(pointPropType).isRequired,
+  ruler: rulerPropType.isRequired,
+};
+
+const labelHeight = 24;
+
+const calculateY = (points, yTooltipPosition) => {
+  const pointsObject = {};
+
+  const pointsSorted = [...points].sort((a, b) => a.y - b.y);
+
+  let prevPoint;
+  for (let i = 0; i < pointsSorted.length; i += 1) {
+    const realSpace = yTooltipPosition - pointsSorted[i].y;
+    const neededSpace = (pointsSorted.length - i) * labelHeight;
+
+    // move point upper if no space until the limit available
+    if (realSpace < neededSpace) {
+      const pointPos = yTooltipPosition - neededSpace;
+      // if no upper room available stick it to 0
+      prevPoint = pointPos >= 0 ? pointPos : 0;
+      // move point lower if it overlaps the previous one
+    } else if (prevPoint + labelHeight > pointsSorted[i].y) {
+      prevPoint += labelHeight;
+    } else {
+      prevPoint = pointsSorted[i].y;
+    }
+    pointsObject[pointsSorted[i].name] = prevPoint;
+  }
+
+  return pointsObject;
+};
+
+const Ruler = ({ ruler, points, height, width }) => {
+  const xTooltipMargin = 5;
+  const xTooltipPosition = height - labelHeight - xTooltipMargin;
```

Comment:
Now it's called `xTooltipPosition`? I'm even more confused.

What does this value actually represent? As far as I can tell, it's just a point 29px (`labelHeight + xTooltipMargin`) off the bottom (`height - 29`) of the chart, but I don't know why, or what it's used for?

Reply:
ok, just added a comment